File: levels/01/04-SmallSquareBigDouble/solution.test.ts
import { expect, test } from 'bun:test';
import { Solution } from './solution';

function fakeGen() {
  const result: any[] = [];
  const fake = {
    log: (a: any) => {
      console.log(a);
      result.push(a);
    },
  };
  return [fake, result];
}

test('01 01 FizzBuzz', () => {
  const [fake, result] = fakeGen();
  Solution(fake);
  expect(result).toEqual([
    49, 64, 81, 100, 121, 144, 169, 196, 225, 256, 289, 324, 361, 40, 42, 44, 46, 48, 50, 52, 54, 56, 58, 60, 62, 64,
    66, 68, 70,
  ]);
});

File: levels/01/04-SmallSquareBigDouble/solution_test.py
from solution import solution
import unittest


def print_gen():
    result = []
    def p(a):
        result.append(a)
        print(a)
    return [p, result]


class TestStringMethods(unittest.TestCase):

    def test_solution(self):
        [fake, result] = print_gen()
        solution(fake)
        expected = [49,64,81,100,121,144,169,196,225,256,289,324,361,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70]
        self.assertEqual(result, expected)


if __name__ == '__main__':
    unittest.main()